lib/cyclonedx/report_ruby_gems_cyclonedx.rb

```diff
@@ -3,5 +3,19 @@ class ReportRubyGems < Base
     def initialize(scan_report)
       super(scan_report)
     end
+
+    def package_url(dependency)
```

Comment:
For Ruby we can use the `dependency[:type]` but for others (i.e node) we'll need to change the type to match with the specs https://github.com/package-url/purl-spec